tests/sys/psa_crypto_ecdsa/Makefile

```diff
@@ -10,11 +10,9 @@ USEMODULE += psa_hash_sha_256
 USEMODULE += psa_asymmetric
 USEMODULE += psa_asymmetric_ecc_p256r1
 
-ifneq (1, $(SHOULD_RUN_KCONFIG))
-  CFLAGS += -DCONFIG_PSA_ASYMMETRIC_KEYPAIR_COUNT=1
-  CFLAGS += -DCONFIG_PSA_SINGLE_KEY_COUNT=1
-endif
+CFLAGS += -DCONFIG_PSA_ASYMMETRIC_KEYPAIR_COUNT=1
+CFLAGS += -DCONFIG_PSA_SINGLE_KEY_COUNT=1
 
-CFLAGS += -DTHREAD_STACKSIZE_MAIN=4096
+CFLAGS += -DTHREAD_STACKSIZE_MAIN=7000
```

Comment:
Was this on purpose? If so, what caused it to require a bigger stack?

(I guess) this is the reason for the following necessary changes to the `Makefile.ci`:
```diff
diff --git a/tests/sys/psa_crypto_ecdsa/Makefile.ci b/tests/sys/psa_crypto_ecdsa/Makefile.ci
index 6a2fca3caf..a0cc10bacb 100644
--- a/tests/sys/psa_crypto_ecdsa/Makefile.ci
+++ b/tests/sys/psa_crypto_ecdsa/Makefile.ci
@@ -1,14 +1,27 @@
 BOARD_INSUFFICIENT_MEMORY := \
     arduino-duemilanove \
     arduino-leonardo \
+    arduino-mega2560 \
     arduino-nano \
     arduino-uno \
     atmega328p \
     atmega328p-xplained-mini \
     atmega8 \
+    bluepill-stm32f030c8 \
+    i-nucleo-lrwan1 \
+    nucleo-f030r8 \
     nucleo-f031k6 \
+    nucleo-f042k6 \
     nucleo-l011k4 \
+    nucleo-l031k6 \
+    nucleo-l053r8 \
     samd10-xmini \
+    slstk3400a \
     stk3200 \
     stm32f030f4-demo \
+    stm32f0discovery \
+    stm32g0316-disco \
+    stm32l0538-disco \
+    waspmote-pro \
+    weact-g030f6 \
     #
```

Reply:
Well, I tested this on the nrf52840dk with the hardware driver and it uses around 6KB of stack.
But it's a worst case. 
There are multiple options:
1. Change thew default backend for this test to micro-ecc (only 1.5 KB stack use). Drawback: we can't test the automatic driver selection with this, because it will never look for available hardware drivers.
2. Only increase the stack for this board. Drawback: We'll need to do this for all boards with hardware drivers, that need more stack than micro-ecc.
3. This is a runtime problem. So far we considered these to be configuration tests. We keep that mindset and just never execute them. Drawback: People might use this as an example for their own applications and ignoring the stack size issues is misleading.
Any opinions on this?

Reply:
> Well, I tested this on the nrf52840dk with the hardware driver and it uses around 6KB of stack.

Ah I see, so it is actually not usable with the cryptocell backend right now on master, and this fixes this.

I think I tend towards 2 because it is the cleanest solution, but with the following suggestion: Could we increase the stack dependent on the selected backend? Not sure whether the Make build system allows us to do this, but maybe by defining the stack size only after the common include?

If not possible, I would increase the stacksize in general as you did, but maybe add a comment stating this is only necessary for the cryptocell backend.

Reply:
```diff
diff --git a/tests/sys/psa_crypto_ecdsa/Makefile b/tests/sys/psa_crypto_ecdsa/Makefile
index 0eb1e488a4..33ed352221 100644
--- a/tests/sys/psa_crypto_ecdsa/Makefile
+++ b/tests/sys/psa_crypto_ecdsa/Makefile
@@ -13,6 +13,12 @@ USEMODULE += psa_asymmetric_ecc_p256r1
 CFLAGS += -DCONFIG_PSA_ASYMMETRIC_KEYPAIR_COUNT=1
 CFLAGS += -DCONFIG_PSA_SINGLE_KEY_COUNT=1
 
-CFLAGS += -DTHREAD_STACKSIZE_MAIN=7000
-
 include $(RIOTBASE)/Makefile.include
+
+ifneq (,$(filter psa_asymmetric_ecc_p256r1_backend_microecc,$(USEMODULE)))
+  CFLAGS += -DTHREAD_STACKSIZE_MAIN=4096
+endif
+
+ifneq (,$(filter psa_asymmetric_ecc_p256r1_backend_periph,$(USEMODULE)))
+  CFLAGS += -DTHREAD_STACKSIZE_MAIN=7000
+endif
```

seems to work for me, but it's not specific to a certain peripheral implementation. What do you think?

Reply:
Updated! Thanks for the suggestion.